### PR TITLE
roles: hosted_engine_setup: Pause on failed restore

### PR DESCRIPTION
**Pull request by didib**
Bug-Url: https://bugzilla.redhat.com/1991171

### CODE REVIEW

**Comment by didib** [comment]
Didn't verify yet

**Comment by didib** [comment]
Verified. Took a backup from latest master, tried to restore on 4.4.7, got:

[ INFO  ] TASK [ovirt.ovirt.hosted_engine_setup : Run engine-backup]
[ INFO  ] engine-backup --mode=restore failed:
         FATAL: Backup was created by version '4.4.8.4_master' and can not be restored using the installed version 4.4.7.6
         You can now connect from this host to the bootstrap engine VM using ssh as root and the temporary IP address - 192.168.1.83 - and fix this issue. Please continue only after the back
up is restored.
         To retry the command that failed, you can run, on the engine VM:
         engine-backup --mode=restore --file=/root/engine_backup --provision-all-databases --restore-permissions
[ INFO  ] TASK [ovirt.ovirt.hosted_engine_setup : include_tasks]
[ INFO  ] ok: [localhost]
[ INFO  ] TASK [ovirt.ovirt.hosted_engine_setup : Create temporary lock file]
[ INFO  ] changed: [localhost]
[ INFO  ] TASK [ovirt.ovirt.hosted_engine_setup : Pause execution until /tmp/ansible.6pl7h7_6_he_setup_lock is removed, delete it once ready to proceed]

Then ssh'ed into the engine vm, updated the engine, and removed the lock file, and deploy continued.

File: changelogs/fragments/327-pause-on-restore.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+minor_changes:
+  - Pause hosted-engine deployment on failure of engine-backup --mode=restore (https://github.com/oVirt/ovirt-ansible-collection/pull/327).
```
Comment (arachmani) [inline changelogs/fragments/327-pause-on-restore.yml:3]: This should start with the role name (for example https://github.com/oVirt/ovirt-ansible-collection/tree/master/changelogs#example)

File: roles/hosted_engine_setup/tasks/restore_backup.yml [hunk 2]
```diff
@@ -13,7 +13,28 @@
     --file=/root/engine_backup --provision-all-databases --restore-permissions
   environment: "{{ he_cmd_lang }}"
   register: engine_restore_out
+  ignore_errors: true
   changed_when: true
+- name: Pause the execution to let the user interactively handle restore failures
+  block:
+    - name: Let the user connect to the bootstrap engine to manually handle restore failures
```
Comment (arachmani) [inline roles/hosted_engine_setup/tasks/restore_backup.yml:20]: Let's be consistent and use `bootstrap engine VM`

File: roles/hosted_engine_setup/tasks/restore_backup.yml [hunk 3]
```diff
@@ -13,7 +13,28 @@
...
+          engine-backup --mode=restore failed:
+
+          {{ engine_restore_out.stderr }}
+
+          You can now connect from this host to the bootstrap engine VM using ssh as root
+          and the temporary IP address -
+          {{ hostvars[he_ansible_host_name]['local_vm_ip']['stdout_lines'][0] }} -
+          and fix this issue. Please continue only after the backup is restored.
+
+          To retry the command that failed, you can run, on the engine VM:
```
Comment (arachmani) [inline roles/hosted_engine_setup/tasks/restore_backup.yml:32]: Ditto

File: changelogs/fragments/327-pause-on-restore.yml (hunk 1 above)
Comment (didib) [reply changelogs/fragments/327-pause-on-restore.yml]: Done

File: roles/hosted_engine_setup/tasks/restore_backup.yml (hunk 2 above)
Comment (didib) [reply roles/hosted_engine_setup/tasks/restore_backup.yml]: Done. We are not that consistent elsewhere, though :-(.

File: roles/hosted_engine_setup/tasks/restore_backup.yml (hunk 3 above)
Comment (didib) [reply roles/hosted_engine_setup/tasks/restore_backup.yml]: Done

**Review by arachmani — APPROVED** [review]

**Review by mnecas — APPROVED** [review]
LGTM

**Comment by michalskrivanek** [comment]
lgtm

**Review by mwperina — COMMENTED** [review]
+1

**Pull request merged**